stdio/file.c

```diff
@@ -579,39 +601,85 @@ char *fgets(char *str, int n, FILE *stream)
 }
 
 
-int fflush_unlocked(FILE *stream)
+static int __fflush_one(FILE *stream)
 {
-	if (stream == NULL) {
-		fflush_unlocked(stdout);
-		return 0; /* TODO: flush all FILE's */
+	int ret = 0;
+
+	if ((stream->flags & F_WRITING) != 0) {
+		if (stream->bufpos != 0) {
```

Comment:
Why not merge this into one `if`?

Reply:
I don't want to go to `else` if `stream->bufpos == 0`

Reply:
I don't want to go to else if stream->bufpos == 0

Reply:
okay, ofc